packages/synapse-react-client/src/lib/containers/discussion_forum/ForumThreadEditor.tsx

```diff
@@ -11,18 +12,32 @@ import {
 } from '../../utils/hooks/SynapseAPI/forum/useReply'
 import { CreateDiscussionThread } from '../../utils/synapseTypes/DiscussionBundle'
 import { MarkdownEditor } from '../markdown/MarkdownEditor'
+import { ConfirmationButtons, ConfirmationDialog } from '../ConfirmationDialog'
 
 export type ForumThreadEditorProps = {
   initialTitle?: string
   initialText?: string
   id: string
   onClose: () => void
   isReply: boolean
-}
+} & (
+  | { isDialog: false; openDialog?: never }
+  | { isDialog: true; openDialog: boolean }
+)
```

Comment:
If `isDialog` is false, then don't allow `openDialog` to be defined. If `isDialog` is true, require that `openDialog` is defined.